src/utility/storage/storage.service.ts

```diff
@@ -81,6 +94,40 @@ export class StorageService {
     });
   }
 
+  async createMultipartUpload(params: Omit<CreateMultipartUploadCommandInput, 'Bucket'>) {
+    const command = new CreateMultipartUploadCommand({ Bucket: this.awsS3BucketStorage, ...params });
+    return this.s3(this.awsS3RegionStorage).send(command);
+  }
+
+  async completeMultipartUpload(params: Omit<CompleteMultipartUploadCommandInput, 'Bucket'>) {
+    const command = new CompleteMultipartUploadCommand({ Bucket: this.awsS3BucketStorage, ...params });
+    return this.s3(this.awsS3RegionStorage).send(command);
+  }
+
+  getSignedUrlForUploadPartStorage(params: Omit<UploadPartCommandInput, 'Bucket'>, expiresIn: number): Promise<string> {
+    const command = new UploadPartCommand({ Bucket: this.awsS3BucketStorage, ...params });
+    return getSignedUrl(this.s3(this.awsS3RegionStorage), command, { expiresIn });
+  }
+
+  async insertVideoAttachment(
+    appId: string,
+    authorId: string,
+    attachmentId: string,
+    file: { name: string; type: string; size: number },
+  ) {
+    const AttachmentRepo = this.entityManager.getRepository(Attachment);
+    return await AttachmentRepo.insert({
+      id: attachmentId,
+      appId,
+      author: { id: authorId },
+      name: file.name,
+      filename: file.name,
+      contentType: file.type,
+      size: file.size,
+      options: [{ source: 's3' }],
+    });
+  }
```

Comment:
Method `insertVideoAttachment` shall write in MediaInfra & wrap in @ MediaService.  
MediaService should import @ `StorageController`'s constructor and invoke @ there.

Reply:
fixed